src/asm2wasm.h

```diff
@@ -509,6 +509,62 @@ class Asm2WasmBuilder {
   }
 
   // Some binary opts might trap, so emit them safely if we are precise
+
+  Expression* makeDangerousI32Binary(BinaryOp op, Expression* left, Expression* right) {
```

Comment:
I find the word "dangerous" that is used here is not that self-documenting, I find I need to read the contents carefully to understand what it means. "Dangerous" from whose perspective? I.e. is it dangerous that the generated code might trap, or dangerous in the sense of unsafe from the user perspective that it will silently ignore div by zero and output an undefined result. Perhaps `makeDangerousI32Binary` could instead be `makePotentiallyTrappingI32Binary` or something like that?

(Not critical at all though, fine with the current name, since this is more an internal code thing, and not something that end users will interact with)